rpc-interface/src/consensus.rs

```diff
@@ -114,7 +114,6 @@ pub trait ConsensusInterface {
         htlc_recipient: Address,
         hash_root: AnyHash,
         hash_count: u8,
-        hash_algorithm: HashAlgorithm,
```

Comment:
How is this RPC method going to work? Does the `hash_root` field now have to be prefixed by a byte representing the algo, so needs to be 33 bytes long?

Reply:
No, this should work the same way we send `enums`. A tag field must be inserted to specify the `hashAlgorithm`.

Reply:
OK, I tested this. And while it's not very ergonomic I think, it does work.

Still, for the `preimage` setting the hash algo doesn't make much sense usually.

Reply:
@sisou Moved the `pre_image` to a separate type